packages/flutter_tools/test/src/throwing_pub.dart

```diff
@@ -46,4 +46,10 @@ class ThrowingPub implements Pub {
   }) {
     throw UnsupportedError('Attempted to invoke pub during test.');
   }
+
```

Comment:
nit: since all of these methods throw, I think we can simplify this class to:

```dart
class ThrowingPub extends Fake implements Pub {}
```

A `Fake` will throw an `UnimplementedError` for any method that is not explicitly implemented.